### PR TITLE
Refactor project form

### PR DESCRIPTION
**Pull request by StephanH90**
Implements the comments left on this PR: https://github.com/adfinis/ember-ebau-gwr/pull/330

### CODE REVIEW

File: addon/components/search.hbs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{{#let
+  (component
+    "search/result-table"
+    results=this.searchResults
+    loading=this.search.isRunning
+  )
+  as |Results|
+}}
+
+{{yield Filters Results}}
```
Comment (velrest) [inline addon/components/search.hbs:19]:
```suggestion
{{yield
  (component
    "search/filters"
    changeset=this.changeset
    onSubmit=this.onSubmit
    validateBeforeSubmit=true
  )
  (component
    "search/result-table"
    results=this.searchResults
    loading=this.search.isRunning
  )
}}
```

File: addon/components/search.hbs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+{{#if this.hasMoreResults}}
+  <Search::ResultTable::LoadMore
+    @onClick={{(perform this.loadMore)}}
+    @loading={{this.search.isRunning}}
+  />
+{{/if}}
+{{/let}}
+{{/let}}
```
Comment (velrest) [inline addon/components/search.hbs:28]:
```suggestion
```

File: addon/components/search.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+export default class SearchComponent extends Component {
+  @tracked page = 0;
+  @tracked searchResults = [];
+  @tracked hasMoreResults = false;
+
+  @service notification;
+  @service intl;
+  @service building;
+  @service street;
```
Comment (velrest) [inline addon/components/search.js:16]: Not needed i think?

File: addon/components/search/filters.hbs [hunk 4]
```diff
@@ -141,5 +147,4 @@
     </div>
   {{/let}}
   {{/let}}
```
Comment (velrest) [inline addon/components/search/filters.hbs:149]:
```suggestion
```

File: addon/components/search/filters.hbs [hunk 5]
```diff
@@ -11,39 +11,45 @@
...
-      disabled=(or this.isSaving @disabled this.isSubmitBtnDisabled)
-      loading=this.submit.isRunning
+      disabled=(or this.isSaving @disabled this.isSubmitDisabled)
+      loading=@onSubmit.isRunning
       buttonComponent=(component "model-form/submit" color="primary")
     )
     as |Submit|
   }}
-  {{!-- Error Select Component. Allows to filter by "hasError" --}}
   {{#let
```
Comment (velrest) [inline addon/components/search/filters.hbs:20]: Move both definitions in one let

File: addon/components/search/filters.js [hunk 6]
```diff
@@ -21,50 +18,16 @@ export default class SearchFiltersComponent extends Component {
...
-  }
-
-  get isSubmitBtnDisabled() {
+  get isSubmitDisabled() {
     return !this.hasChanges;
   }
 
   get hasChanges() {
-    return !!this.changeset.changes.find(({ value }) => value !== "");
+    return !!this.args.changeset.changes.find(({ value }) => value !== "");
```
Comment (velrest) [inline addon/components/search/filters.js:26]: Nitpick for better readability:
```suggestion
    return Boolean(this.args.changeset.changes.find(({ value }) => value !== ""));
```

File: addon/components/search/result-table.hbs [hunk 7]
```diff
@@ -1,8 +1,11 @@
-<table
-  class="uk-table uk-table-middle uk-table-divider uk-table-striped"
-  ...attributes
->
-  {{yield
-    (component "search/result-table/row" results=@results loading=@loading)
-  }}
-</table>
+{{#if (gt @results.length 0)}}
```
Comment (velrest) [inline addon/components/search/result-table.hbs:1]: Nitpick: length is falsy if 0 so `{{#if @result.length}}` would be enough .
```suggestion
{{#if (gt @results.length 0)}}
```

File: addon/controllers/building/edit/entrance/edit/link-street.js [hunk 8]
```diff
@@ -21,16 +21,19 @@ export default class BuildingEditEntranceLinkStreetController extends Controller
   @tracked PLZ4 = null;
   @tracked PLZ6 = null;
 
-  get baseQuery() {
+  get baseModel() {
```
Comment (velrest) [inline addon/controllers/building/edit/entrance/edit/link-street.js:24]: Why did you rename it? I think the previous name was more accurate since its not really a model.

File: addon/components/search.hbs (hunk 1 above)
Comment (StephanH90) [reply addon/components/search.hbs]: done

File: addon/components/search.hbs (hunk 2 above)
Comment (StephanH90) [reply addon/components/search.hbs:33]: done

File: addon/components/search.js (hunk 3 above)
Comment (StephanH90) [reply addon/components/search.js:16]: done

File: addon/components/search/filters.hbs (hunk 5 above)
Comment (StephanH90) [reply addon/components/search/filters.hbs:20]: done

File: addon/components/search/filters.js (hunk 6 above)
Comment (StephanH90) [reply addon/components/search/filters.js:26]: done

File: addon/components/search/result-table.hbs (hunk 7 above)
Comment (StephanH90) [reply addon/components/search/result-table.hbs]: has been removed

File: addon/controllers/building/edit/entrance/edit/link-street.js (hunk 8 above)
Comment (StephanH90) [reply addon/controllers/building/edit/entrance/edit/link-street.js]: done

File: addon/components/search/filters.hbs (hunk 5 above)
Comment (velrest) [reply addon/components/search/filters.hbs:20]: Did you push your latest changes? still looks the same.

**Review by velrest — APPROVED** [review]

**Review by velrest — DISMISSED** [review]

**Review by velrest — APPROVED** [review]
